### PR TITLE
Bug/rerender fixes

### PR DESCRIPTION
**Pull request by lund0n**
## Change Type

* [ ] Feature
* [ ] Chore
* [x] Bug Fix

## Change Level

* [ ] major
* [ ] minor
* [x] patch

## Further Information (screenshots, etc)

This PR fixes two primary issues:

1. `boundingLeft` and `boundingRight` props were lost from in-view checks when props were updated in the component.
2. The DOM node that was evaluated was cached in `componentDidMount`. This meant that if the root node changed for any reason that the boundary checks would not work correctly.

Additionally, I:

* Switched to `react-testing-library`
* Fixed the code coverage directories.
* Updated contributors.

## Checklist

* [x] Added tests / did not decrease code coverage
* [ ] Tested across browsers

### CODE REVIEW

**Comment by zacanger** [comment]
Is `react-testing-library` the new hotness?

File: package.json [hunk 1]
```diff
@@ -113,4 +111,4 @@
     "react": ">=16.0.0",
     "react-dom": ">=16.0.0"
   }
-}
+}
```
Comment (zacanger) [inline package.json]: `\n` here

**Comment by lund0n** [comment]
@zacanger It's definitely simpler to work with. It's just a wrapper around `react-dom` and `react-dom/test-utils`. I like that it prevents you from accessing the component internals, and gets it closer to interacting with the DOM itself.

File: package.json (hunk 1 above)
Comment (lund0n) [reply package.json]: Weird. I didn't add the the `\n`. Maybe when installed/uninstalled packages it added it.

**Pull request merged**